en/Data Structures/Queues/queue.md

```diff
@@ -0,0 +1,49 @@
+# Queue
+
+## Description:
+A queue is a linear data structure that follows the First-In-First-Out (FIFO) principle. It is often compared to a real-world queue of people waiting in line. The element that is added first is the one that gets removed first. Queues are commonly used for various applications, such as task scheduling, managing requests, and more.
+
+### Time Complexity:
+Enqueue (Insertion): O(1)
```

Comment:
Format this as a list please. Otherwise this will render as one paragraph, which is undesirable.